pgconn/pgconn.go

```diff
@@ -896,17 +896,11 @@ func (pgConn *PgConn) CancelRequest(ctx context.Context) error {
 	binary.BigEndian.PutUint32(buf[4:8], 80877102)
 	binary.BigEndian.PutUint32(buf[8:12], uint32(pgConn.pid))
 	binary.BigEndian.PutUint32(buf[12:16], uint32(pgConn.secretKey))
+	// Postgres will process the request and close the connection
+	// so when don't need to read the reply
+	// https://www.postgresql.org/docs/current/protocol-flow.html#id-1.10.6.7.10
 	_, err = cancelConn.Write(buf)
-	if err != nil {
-		return err
-	}
-
-	_, err = cancelConn.Read(buf)
-	if err != io.EOF {
-		return err
-	}
-
-	return nil
+	return err
```

Comment:
This will change behavior somewhat. Read until `EOF` means that `CancelRequest` will only return after the server has definitely received the cancel message or an error occurs. With this change it will return without waiting for the acknowledgement of the server closing the connection.

But I'm not sure if the change is meaningful as the server side cancel request is asynchronous anyway. 🤔

Reply:
> This will change behavior somewhat. Read until `EOF` means that `CancelRequest` will only return after the server has definitely received the cancel message or an error occurs. With this change it will return without waiting for the acknowledgement of the server closing the connection.
> 
> But I'm not sure if the change is meaningful as the server side cancel request is asynchronous anyway. thinking

once write returns we have sent the request to Postgres. It should be safe to stop here. 
If you prefer I catch the `windows.WSAECONNRESET` as @codercms suggested above